source/cpu/cpu.go

```diff
@@ -276,9 +282,38 @@ func discoverTopology() map[string]string {
 		features["hardware_multithreading"] = strconv.FormatBool(ht)
 	}
 
+	if socketCount, err := getCPUSocketCount(); err != nil {
+		klog.ErrorS(err, "failed to get sockets count")
+	} else {
+		features["socket_count"] = strconv.FormatInt(socketCount, 10)
+	}
+
 	return features
 }
 
+func getCPUSocketCount() (int64, error) {
+	files, err := os.ReadDir(hostpath.SysfsDir.Path("bus/cpu/devices"))
+	if err != nil {
+		return 0, err
+	}
+
+	uniquePhysicalIDs := make(map[int]bool)
+
+	for _, file := range files {
+		// Try to read physical_package_id from topology
+		physicalID, err := os.ReadFile(hostpath.SysfsDir.Path("bus/cpu/devices", file.Name(), "topology/physical_package_id"))
+		if err != nil {
+			return 0, err
+		}
+		id, err := strconv.Atoi(strings.TrimSuffix(string(physicalID), "\n"))
```

Comment:
We don't actually need to convert the value to integer type, I think. Then, we could use TrimSpace to simplify a bit (trima all leading and trailing whitespace).

Thus, how about?


```suggestion
		id, err := strings.TrimSpace(string(physicalID))
```